jetty-util/src/main/java/org/eclipse/jetty/util/BlockingCallback.java

```diff
@@ -0,0 +1,82 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2021 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.util;
+
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Provides a {@link Callback} that can block the thread
+ * while waiting to be completed.  The block occurs in the
+ * auto-closing of the callback.
+ * <p>
+ * A typical usage pattern is:
+ * <pre>
+ * void someBlockingCall(Object... args) throws IOException
+ * {
+ *     try(BlockingCallback blocker = new BlockingCallback())
+ *     {
+ *         someAsyncCall(args, blocker);
+ *     }
+ * }
+ * </pre>
+ * @see SharedBlockingCallback
+ */
+public class BlockingCallback implements AutoCloseable, Callback, Runnable
+{
+    private static final Throwable SUCCEEDED = new Throwable();
```

Comment:
Maybe override `fillInStackTrace` as the stack trace is never going to be needed?